frontend/actions/documentsActions.js

```diff
@@ -33,6 +33,41 @@ export function deleteDocuments ({api, collection, ids}) {
   }
 }
 
+export function deleteMedia ({api, collection, ids}) {
+  return (dispatch, getState) => {
+    Promise.all(
+      ids.map(id =>
+        fetch(
+          `${api.host}:${api.port}/media/${id}`,
+          {
+            headers: {
+              'Accept': 'application/json',
+              'Authorization': `Bearer ${getState().user.accessToken}`
+            },
+            method: 'DELETE'
+          }
+        )
+      )
+    )
+    .then(
+      responses => {
+        dispatch(
```

Comment:
You're dispatching the "is deleting" action after `Promise.all` resolves, at which point all items have been deleted?

Reply:
Ah, so I should use that first, and then when it resolves use a "has deleted", right?

Reply:
See the [deleteDocuments action](https://github.com/dadi/publish/blob/2061a86bd28d1452a4d7d7c12f5052104c99c2ca/frontend/actions/documentsActions.js#L15-L34) example. We set the state to "saving" before we start the operation. The action that is fired after the operation is complete will make the reducer set `isDeleting` back to false, which the components can listen to to know when a delete operation has finished (e.g. `previousState.isDeleting && !currentState.isDeleting`).